cmd/kube-proxy/app/server.go

```diff
@@ -230,7 +260,47 @@ func (o *Options) Run() error {
 		return err
 	}
 
-	return proxyServer.Run()
+	errCh := make(chan error)
+	defer close(errCh)
+
+	// run the proxy in goroutine
+	go func() {
+		var isRunning bool
+		for {
+			select {
+			case <-o.stopCh:
+				return
+			default:
+				if !isRunning {
+					isRunning = true
+					go func() {
+						err := proxyServer.Run()
```

Comment:
if Run() ever returns, this will be a hotloop, won't it?

stopCh has nothing, isRunning  is true, loop as fast as possible.

This feels more complicated than it needs to be.  Can't we just assume that after OnChange() is called this whole function is going to exit?  What are you designing for here?

Reply:
@thockin Updated. PTAL. Thanks.